hudi-common/src/main/java/org/apache/hudi/metadata/MetadataPartitionType.java

```diff
@@ -18,30 +18,52 @@
 
 package org.apache.hudi.metadata;
 
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 /**
  * Partition types for metadata table.
  */
 public enum MetadataPartitionType {
```

Comment:
nit: rename this now to `MetadataPartition` since this is no long about a "type"

Reply:
hmmm. second thought. its still an enum. so take it back. ok with this name for now